### DISCUSSION TITLE
Roadmap Plan for GA

### DISCUSSION THREAD

**Discussion by tafs7** [Azure/azure-functions-sql-extension#695] [General] [answered]
Could someone provide a roadmap for when this is roughly planned for GA?

I'm working with a client who's worried about using a preview feature in Azure, but without this feature, the effort we'd need to put in place is much greater and more complex. Having a rough timeline would be helpful for us to make better informed decisions now.

Thanks!

**Comment by dzsquared** [comment, marked as answer]
The GA for input/output bindings can be expected early in this calendar year.  At the same time we are pressing forward on the SQL trigger and its GA will follow at a later date.


EDIT: yes, the trigger is still in development.  Mentions of "removal" are from the main branch while it remains in a feature branch such that we can release the input/output bindings to GA very soon while completing the trigger.

**Reply by tafs7** [reply]
Awesome. Thanks! From a support perspective, I assume for the trigger feature, as long as it is in preview, it will only have limited support?

**Reply by dzsquared** [reply]
While the trigger feature is in preview, there is reduced support in comparison with the increased support provided through Azure Support and other avenues. The engineering team here is interested in making sure that the trigger, even while in preview, is valuable and we're going to continue to address issues raised here.
The other preview vs GA component I wanted to mention is that while it isn't necessarily likely, it is possible that there is a breaking change make to a component while in preview.

I added an issue to lay out some of the concepts in the roadmap as well - https://github.com/Azure/azure-functions-sql-extension/issues/745

**Comment by denious** [comment]
@dzsquared Can you please comment on the SQL trigger code and documentation recently being entirely removed from the repository? Are we still pressing forward on the SQL trigger release and where will the docs and the code live?

**Reply by dzsquared** [reply]
The "removal" of the trigger code from the main branch is only to support a GA package of input/output bindings - the trigger is still in development (language support is currently in PR https://github.com/Azure/azure-functions-sql-extension/pull/722) and will continue to be released as a preview until its ready for GA.

**Reply by Charles-Gagnon** [reply]
Apologies for the confusion - as Drew mentioned development for trigger will continue in https://github.com/Azure/azure-functions-sql-extension/tree/release/trigger until it's ready for GA!

**Comment by shobhitkasliwal-onbe** [comment]
Do we have any update on SQL Trigger GA ?
